forge-game/src/main/java/forge/game/GameAction.java

```diff
@@ -233,6 +233,13 @@ private Card changeZone(final Zone zoneFrom, Zone zoneTo, final Card c, Integer
             if (!lastKnownInfo.hasKeyword("Counters remain on CARDNAME as it moves to any zone other than a player's hand or library.")) {
                 copied.clearCounters();
             }
+            if (cause != null && cause.isCraft() && toBattlefield) {
+                CardCollection cW = new CardCollection();
+                for (Card cr : cause.getPaidList("Exiled")) {
+                    if (!cr.equals(cause.getHostCard())) cW.add(cr);
+                }
+                copied.setCraftedWith(cW);
```

Comment:
imo we might not need to have extra property for Card (that needs to be cleaned up later)

but we could just reuse copied.setExiledWith ?

@tool4ever @tehdiplomat your opinion about this?

Reply:
I like this idea – I will give it a go

Reply:
Technically ruleswise they should be different, but for us no need to reinvent the wheel

Reply:
Right - if it creates a conflict with the rules later we can always separate it